components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImpl.java

```diff
@@ -140,6 +143,28 @@ public Action getActionByActionId(String actionId, String tenantDomain) throws A
         return CACHE_BACKED_DAO.getActionByActionId(actionId, IdentityTenantUtil.getTenantId(tenantDomain));
     }
 
+    @Override
+    public Action updateActionEndpointAuthentication(String actionType, String actionId, AuthType authentication,
+                                                     String tenantDomain) throws ActionMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Updating Action endpoint authentication for Auth Type: " + authentication.getType().name() +
+                    " and Action Type: " + actionType + " and Action Id: " + actionId);
+        }
+        Action action = checkIfActionExists(actionId, tenantDomain);
+        if (action.getEndpoint().getAuthentication().getType().equals(authentication.getType())) {
+
+            return CACHE_BACKED_DAO.updateActionEndpointAuthProperties(actionId, authentication,
+                    IdentityTenantUtil.getTenantId(tenantDomain));
+        } else {
+            EndpointConfig endpoint = new EndpointConfig.EndpointConfigBuilder()
+                    .uri(action.getEndpoint().getUri())
+                    .authentication(authentication).build();
+            return CACHE_BACKED_DAO.updateActionEndpoint(getActionTypeFromPath(actionType), actionId, endpoint,
+                    action.getEndpoint().getAuthentication(), IdentityTenantUtil.getTenantId(tenantDomain));
+        }
+    }
```

Comment:
Overall on this method it's bit hard to read at first why you have a condition and change auth properties and in the other one change the action fully.
If we can break this into two functions that would convey the message it would be easy to map

Reply:
FIxed with https://github.com/wso2/carbon-identity-framework/pull/5795/commits/4c40292ff2e65531731f8e68897c74215dae5279